dictionary/views.py

```diff
@@ -1,3 +1,27 @@
+'''
+Views for the dictionary application.
+'''
+
 from django.shortcuts import render
+from django.http import HttpResponseRedirect
+from django.contrib.postgres.search import SearchVector
+
+from . import forms
+from . import models
+
+def search(request):
+    '''
+    This is a first crack at a basic search interface.
+    '''
+    if request.method == 'POST':
+        form = forms.SearchForm(request.POST)
+        if form.is_valid():
+            search_text = form.cleaned_data['search_text']
+            search_vector = SearchVector('simple', 'traditional', 'pronunciation')
+            results = models.Entry.objects.annotate(search=search_vector).filter(search=search_text)
+            form = forms.SearchForm()
+            return render(request, 'dictionary/search.html', {'form': form, 'results': results,})
+    else:
+        form = forms.SearchForm()
```

Comment:
So to continue our discussion about the GET method, the code here would have to account for the fact that the method is always GET, even when the form isn't being submitted. So it would look something like this:

```python
if request.GET: # Check if the GET QueryDict is non-empty
    form = forms.SearchForm(request.GET)
    if form.is_valid():
        ...
else:
    form = forms.SearchForm() 
```